src/server/scripts/Spells/spell_priest.cpp

```diff
@@ -573,85 +573,64 @@ class spell_pri_mind_bomb : public AuraScript
 };
 
 // 47540 - Penance
-class spell_pri_penance : public SpellScriptLoader
+class spell_pri_penance : public SpellScript
 {
-    public:
-        spell_pri_penance() : SpellScriptLoader("spell_pri_penance") { }
-
-        class spell_pri_penance_SpellScript : public SpellScript
-        {
-            PrepareSpellScript(spell_pri_penance_SpellScript);
-
-            bool Load() override
-            {
-                return GetCaster()->GetTypeId() == TYPEID_PLAYER;
-            }
+    PrepareSpellScript(spell_pri_penance);
 
-            bool Validate(SpellInfo const* spellInfo) override
-            {
-                SpellInfo const* firstRankSpellInfo = sSpellMgr->GetSpellInfo(SPELL_PRIEST_PENANCE_R1, DIFFICULTY_NONE);
-                if (!firstRankSpellInfo)
-                    return false;
-
-                // can't use other spell than this penance due to spell_ranks dependency
-                if (!spellInfo->IsRankOf(firstRankSpellInfo))
-                    return false;
+    bool Validate(SpellInfo const* /*spellInfo*/) override
+    {
+        return ValidateSpellInfo
+        ({
+            SPELL_PRIEST_PENANCE_DAMAGE,
+            SPELL_PRIEST_PENANCE_HEALING
+        });
+    }
 
-                uint8 rank = spellInfo->GetRank();
-                if (!sSpellMgr->GetSpellWithRank(SPELL_PRIEST_PENANCE_R1_DAMAGE, rank, true))
-                    return false;
-                if (!sSpellMgr->GetSpellWithRank(SPELL_PRIEST_PENANCE_R1_HEAL, rank, true))
-                    return false;
+    bool Load() override
+    {
+        return GetCaster()->GetTypeId() == TYPEID_PLAYER;
```

Comment:
why are we checking if its a player if we nowhere explicitly use player?

Reply:
You actually do on SpellCastResult CheckCast(), maybe the target has any immune_to_pc and shouldn't be eligible for the spell since the caster is a player.

Reply:
That sounds weird and doesnt make any sense. If I cast as a player and the target shouldnt be affected by a spell it required IMMUNE_TO_PC, if i cast it as a npc it requires IMMUNE_TO_NPC, theres nothing related to spellscripts at this point.
We only have to check if the caster is a player if we actually use stuff from the `Player` class itself and dont want to check it right before casting. And we do not use anything from Player in this script, just from inherited classes